File: src/test/scala/ROT13Test.scala
import org.scalatest._

class ROT13Test extends FlatSpec with Matchers {
  final val cipher = new ROT13

  "ROT13" should "extend IROT13" in {
    cipher.isInstanceOf[IROT13] shouldEqual true
  }

  "Encrypt" should "validate" in {
    cipher.encrypt("Uryyb Jbeyq!") shouldEqual "Hello World!"
  }
}

File: rot13_test.go
package rot13

import "testing"

func TestROT13Transform(t *testing.T) {
	if ROT13Transform("Uryyb Jbeyq!") != "Hello World!" {
		t.Errorf("found '%v' expected 'Hello World!'", ROT13Transform("Uryyb Jbeyq!"))
	}
}